src/GloryDuckTables2019Lkl.cc

```diff
@@ -207,51 +207,73 @@ Int_t GloryDuckTables2019Lkl::ReadGloryDuckInputData(TString filename)
   else
     cout << "GloryDuckTables2019Lkl::ReadGloryDuckInputData (" << GetName() << ") Message: Reading masses and <sigmav> values from file " << filename << endl;
 
-  vector<Double_t> vsigma ;    // vector with <sv> values
-  vector<Double_t> vlkl ;      // vecot with -2logL values
-
-  // get first line of the file (with <sv> values)
-  string sigmavLine;
-  getline(ff,sigmavLine);
+  // get the number of lines of the file
+  Int_t numRows = 0;
+  string line;
+  while (getline(ff, line))
+    numRows++;
+  ff.clear();
+  ff.seekg(0, ios::beg);
+
+  vector<Double_t> sigmav ;                   // vector with <sv> values
+  vector<Double_t> mass ;                     // vector with mass values
+  vector<vector<Double_t> > vlkl2D(numRows-1);// 2D vector with -2logL values
+  vector<Double_t> vlkl;                      // vector with -2logL values
+
+  // get first line of the file (with mass values)
+  string massLine;
+  getline(ff,massLine);
 
   // read first line of the file
-  std::istringstream row(sigmavLine);
+  std::istringstream first_line(massLine);
   Double_t field;
-  while (row>> field)
+  while (first_line >> field)
     {
+      if(fNsvVals != 0)
```

Comment:
Rather than skipping the first cell (logJ) you should store it in a private variable of the class so that later we can check that all logJ values from a same source are the same.